saleor/graphql/product/mutations/products.py

```diff
@@ -1785,6 +1788,45 @@ def perform_mutation(cls, _root, info, product_id, images_ids):
         return ProductImageReorder(product=product, images=images)
 
 
+class ProductVariantSetDefault(BaseMutation):
+    product = graphene.Field(Product)
+
+    class Arguments:
+        product_id = graphene.ID(
+            required=True,
+            description="Id of a product that will have the default variant set.",
+        )
+        variant_id = graphene.ID(
+            required=True, description="Id of a variant that will be set as default.",
+        )
+
+    class Meta:
+        description = (
+            "Set default variant for a product. "
+            "Mutation triggers PRODUCT_UPDATED webhook."
+        )
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = ProductError
+        error_type_field = "product_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, product_id, variant_id):
+        product = cls.get_node_or_error(
+            info, product_id, field="product_id", only_type=Product
+        )
+        variant = cls.get_node_or_error(
+            info,
+            variant_id,
+            field="variant_id",
+            only_type=ProductVariant,
+            qs=product.variants.all(),
+        )
+        product.default_variant = variant
```

Comment:
I would add validation that checks if the variant belongs to the product, as now you can set any variant from the system.

Reply:
You cannot. See that the variant has `qs=product.variants.all()`, there is also a test for that `test_product_variant_set_default_not_products_variant`.

Reply:
Ahh missed that, nevermind then :)